playjson/src/main/scala/io/finch/playjson/package.scala

```diff
@@ -11,10 +14,16 @@ package object playjson {
    * @tparam A the type of the data to decode into
    */
   implicit def decodePlayJson[A](implicit reads: Reads[A]): Decode.Json[A] =
-    // TODO: Eliminate toString conversion
-    // See https://github.com/finagle/finch/issues/511
-    // PlayJson can parse from byte[] automatically detecting the charset.
-    Decode.json((b, cs) => Try(Json.parse(BufText.extract(b, cs)).as[A]))
+    Decode.json { (b, cs) =>
+      val buf = ChannelBufferBuf.Owned.extract(b)
+      if (buf.hasArray) {
```

Comment:
Its not a burden. Any particular place we should extract it to? I was thinking `io.finch.Decode`.